idaes/generic_models/flowsheets/methanol_flowsheet.py

```diff
@@ -0,0 +1,396 @@
+    m.fs.M101.H2_WGS.pressure.fix(30e5)  # Pa
+
+    m.fs.M101.CO_WGS.flow_mol[0].fix(316.8)  # mol/s, relative to 88 kmol/h
+    m.fs.M101.CO_WGS.mole_frac_comp[0, "H2"].fix(1e-6)
+    m.fs.M101.CO_WGS.mole_frac_comp[0, "CO"].fix(1)
+    m.fs.M101.CO_WGS.mole_frac_comp[0, "CH3OH"].fix(1e-6)
+    m.fs.M101.CO_WGS.mole_frac_comp[0, "CH4"].fix(1e-6)
+    m.fs.M101.CO_WGS.mole_frac_comp[0, "H2O"].fix(1e-6)
+    m.fs.M101.CO_WGS.enth_mol[0].fix(-110676.4)  # J/mol
+    m.fs.M101.CO_WGS.pressure.fix(30e5)  # Pa
+    print('DOF after streams specified: ', degrees_of_freedom(m))
+
+    # units specifications
+    m.fs.C101.outlet.pressure.fix(51e5)  # Pa
+
+    m.fs.H101.outlet_temp = Constraint(
+        expr=m.fs.H101.control_volume.properties_out[0].temperature
+        == 488.15*pyunits.K)
+
+    m.fs.R101.conversion = Var(initialize=0.75, bounds=(0, 1))
+    m.fs.R101.conv_constraint = Constraint(
+        expr=(m.fs.R101.conversion * m.fs.R101.inlet.flow_mol[0] *
+              m.fs.R101.inlet.mole_frac_comp[0, "CO"] ==
+              m.fs.R101.inlet.flow_mol[0] *
+              m.fs.R101.inlet.mole_frac_comp[0, "CO"]
+              - m.fs.R101.outlet.flow_mol[0] *
+              m.fs.R101.outlet.mole_frac_comp[0, "CO"]))
+    m.fs.R101.conversion.fix(0.75)
+    m.fs.R101.outlet_temp = Constraint(
+        expr=m.fs.R101.control_volume.properties_out[0].temperature
+        == 507.15*pyunits.K)
+    m.fs.R101.heat_duty.setub(0)  # rxn is exothermic, so duty is cooling only
+
+    m.fs.T101.deltaP.fix(-2e6)
+    m.fs.T101.efficiency_isentropic.fix(0.9)
+
+    m.fs.H102.outlet_temp = Constraint(
+        expr=m.fs.H102.control_volume.properties_out[0].temperature
+        == 407.15*pyunits.K)
+
+    m.fs.F101.recovery = Var(initialize=0.01, bounds=(0, 1))
+    m.fs.F101.rec_constraint = Constraint(
+        expr=(m.fs.F101.recovery == m.fs.F101.liq_outlet.flow_mol[0] *
+              m.fs.F101.liq_outlet.mole_frac_comp[0, "CH3OH"] /
+              (m.fs.F101.inlet.flow_mol[0] *
+               m.fs.F101.inlet.mole_frac_comp[0, "CH3OH"])))
+    m.fs.F101.deltaP.fix(0)  # Pa
+    m.fs.F101.outlet_temp = Constraint(
+        expr=m.fs.F101.control_volume.properties_out[0].temperature
+        == 407.15*pyunits.K)
+
+    print('DOF after units specified: ', degrees_of_freedom(m))
+
+
+def initialize_flowsheet(m):
+
+    # Initialize and solve flowsheet
+
+    print('')
+    m.fs.M101.initialize()
+    propagate_state(arc=m.fs.s02)  # mixer to compressor
+
+    m.fs.C101.initialize()
+    propagate_state(arc=m.fs.s03)  # compressor to heater
+
+    m.fs.H101.initialize()
+    propagate_state(arc=m.fs.s04)  # heater to reactor
+
+    m.fs.R101.initialize()
+    propagate_state(arc=m.fs.s05)  # reactor to turbine
+
+    m.fs.T101.initialize()
+    propagate_state(arc=m.fs.s06)  # turbine to cooler
+
+    m.fs.H102.initialize()
+    propagate_state(arc=m.fs.s07)  # cooler to flash
+
+    m.fs.F101.initialize()
+
+
+def add_costing(m):
```

Comment:
This is going to need to be updated before November when the old costing framework is deprecated.